src/access.rs

```diff
@@ -0,0 +1,339 @@
+					.chain(locked_queued_transactions.iter())
+					.cloned()
+					.collect();
+
+				// Remember all registered but unconfirmed transactions for future processing.
+				let mut unconfirmed_registered_txs = Vec::new();
+
+				for txid in registered_txs {
+					if let Some(tx_status) = client.get_tx_status(&txid)? {
+						if tx_status.confirmed {
+							if let Some(tx) = client.get_tx(&txid)? {
+								if let Some(block_height) = tx_status.block_height {
+									let block_header = client.get_header(block_height)?;
+									if let Some(merkle_proof) = client.get_merkle_proof(&txid)? {
+										confirmed_txs.push((
+											tx,
+											block_height,
+											block_header,
+											merkle_proof.pos,
+										));
+										continue;
+									}
+								}
+							}
+						}
+					}
+					unconfirmed_registered_txs.push(txid);
+				}
+
+				// Check all registered outputs for dependent spending transactions.
+				let registered_outputs: Vec<WatchedOutput> = locked_watched_outputs
+					.iter()
+					.chain(locked_queued_outputs.iter())
+					.cloned()
+					.collect();
+
+				// Remember all registered outputs that haven't been spent for future processing.
+				let mut unspent_registered_outputs = Vec::new();
+
+				for output in registered_outputs {
+					if let Some(output_status) = client
+						.get_output_status(&output.outpoint.txid, output.outpoint.index as u64)?
+					{
+						if output_status.spent {
+							if let Some(spending_tx_status) = output_status.status {
+								if spending_tx_status.confirmed {
+									let spending_txid = output_status.txid.unwrap();
+									if let Some(spending_tx) = client.get_tx(&spending_txid)? {
+										let block_height = spending_tx_status.block_height.unwrap();
+										let block_header = client.get_header(block_height)?;
+										if let Some(merkle_proof) =
+											client.get_merkle_proof(&spending_txid)?
+										{
+											confirmed_txs.push((
+												spending_tx,
+												block_height,
+												block_header,
+												merkle_proof.pos,
+											));
+											continue;
+										}
+									}
+								}
+							}
+						}
+					}
+					unspent_registered_outputs.push(output);
+				}
+
+				// Sort all confirmed transactions first by block height, then by in-block
+				// position, and finally feed them to the interface in order.
+				confirmed_txs.sort_unstable_by(
+					|(_, block_height1, _, pos1), (_, block_height2, _, pos2)| {
+						block_height1.cmp(&block_height2).then_with(|| pos1.cmp(&pos2))
+					},
+				);
+				for (tx, block_height, block_header, pos) in confirmed_txs {
+					for c in &confirmables {
+						c.transactions_confirmed(&block_header, &[(pos, &tx)], block_height);
+					}
+				}
+
+				*locked_watched_transactions = unconfirmed_registered_txs;
+				*locked_queued_transactions = Vec::new();
+				*locked_watched_outputs = unspent_registered_outputs;
+				*locked_queued_outputs = Vec::new();
+			}
+
+			{
```

Comment:
i really like that we have separated these out into different blocks, 
can we move them to separate sub-functions as well?

Reply:
Done with https://github.com/lightningdevkit/ldk-lite/pull/9/commits/bdc1b8e5414a41bdee6a501e434260285df4f926.